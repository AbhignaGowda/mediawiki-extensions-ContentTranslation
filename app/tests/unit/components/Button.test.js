import { shallowMount } from "@vue/test-utils";
import MwButton from "@/lib/mediawiki.ui/components/MWButton";

describe("Button.vue", () => {
  it("renders props.id when passed", () => {
    const id = "buttonId";
    const buttonClass = "mw-ui-button";
    const wrapper = shallowMount(MwButton, {
      propsData: { id }
    });
    expect(wrapper.attributes("id")).toMatch(id);
    expect(wrapper.classes(buttonClass)).toBe(true);
  });

  it("matches the snapshot with an HTML element passed as slot", () => {
    const wrapper = shallowMount(MwButton, {
      propsData: {
        type: "progressive"
      },
      slots: {
        default: '<span class="italic">button text</span>'
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it("matches the snapshot with an destructive type", () => {
    const wrapper = shallowMount(MwButton, {
      propsData: {
        type: "destructive"
      },
      slots: {
        default: '<span class="italic">button text</span>'
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
