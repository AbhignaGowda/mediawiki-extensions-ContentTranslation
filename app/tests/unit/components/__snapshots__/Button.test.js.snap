// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue matches the snapshot with an HTML element passed as slot 1`] = `
<button
  class="mw-ui-button"
  type="progressive"
>
  <span
    class="italic"
  >
    button text
  </span>
</button>
`;

exports[`Button.vue matches the snapshot with an destructive type 1`] = `
<button
  class="mw-ui-button"
  type="destructive"
>
  <span
    class="italic"
  >
    button text
  </span>
</button>
`;
